### ISSUE TITLE
NextStaticAssetsS3Bucket - assets already exists.

### ISSUE THREAD

**Issue by romainquellec** [danielcondemarin/serverless-nextjs-plugin#16]
Error in Circle CI : 
```
erverless: Creating Stack...
Serverless: Checking Stack create progress...
......
Serverless: Operation failed!
Serverless: View the full error output: https://eu-west-1.console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stack/detail?stackId=arn%3Aaws%3Acloudformation%3Aeu-west-1%3A899832958734%3Astack%2Fserverless-ssr-production%2F403c64d0-4afa-11e9-aece-061c3112010a
 
  Serverless Error ---------------------------------------
 
  An error occurred: NextStaticAssetsS3Bucket - assets already exists.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information -----------------------------
     OS:                     linux
     Node Version:           8.10.0
     Serverless Version:     1.39.1
```

![image](https://user-images.githubusercontent.com/1150703/54677641-9c40fa80-4b03-11e9-87ac-b19b04da18cf.png)

I deleted the stack, and the bucket doesnt exist before deployment.

**Comment by romainquellec** [comment]
My current next config : 
```
assetPrefix: "https://s3.amazonaws.com/assets",
```
I see in AWS : 
![image](https://user-images.githubusercontent.com/1150703/54677859-1a050600-4b04-11e9-803a-7c4a1982664a.png)

**Comment by danielcondemarin** [comment]
Bucket names [must be unique](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html) globally. I've just checked and "assets" is taken already. You'll need a different bucket name.

**Comment by romainquellec** [comment]
"assets" is taken already.
Thanks ! But taken by what ?

**Comment by danielcondemarin** [comment]
You can add dots in your bucket name which kind of acts like a subdomain, for example:
```js
const nextConfig = {
  assetPrefix: "https://s3.amazonaws.com/assets.cuistotducoin",
```
It wouldn't be related to your main domain though, which is why the bucket provisioned by the plugin allows CORS, so you can host the pages on a different domain.

**Comment by romainquellec** [comment]
So, as an improvements, it should be possible with serverless to create a subdomain, create the bucket, and link the bucket to the subdomain, so we can use it in next ?
If so, can you make this an improvement ?

**Comment by danielcondemarin** [comment]
I'm thinking of adding support for a new plugin config key `assetsBucketName`:

```yml
plugins:
  - serverless-nextjs-plugin

custom:
  serverless-nextjs:
    nextConfigDir: ./
    assetsBucketName: "assets.cuistotducoin.com"
```

Then in your `next.config` you can have:

```js
const config = {
  target: "serverless",
  assetPrefix: "http://assets.cuistotducoin.com"
};
``` 

That way you can have your own subdomain as a CNAME of the bucket as described [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html#VirtualHostingCustomURLs).
It would only work over HTTP though, without SSL.

**Comment by danielcondemarin** [comment]
This is now available on v1.3.1 👍 https://github.com/danielcondemarin/serverless-nextjs-plugin#hosting-static-assets
